scipy/_lib/_c99compat.h

```diff
@@ -6,110 +6,153 @@
+                            return y;
+                        if (!xnan && ynan)
+                            return x;
+                        return x;
+                    }
 
-            /* +0 > -0 is preferred by C the standard */
-            if (x == 0 && y == 0)
-            {
-                int xs = signbit(x), ys = signbit(y);
-                if (xs && !ys)
-                    return y;
-                if (!xs && ys)
-                    return x;
-                return x;
-            }
+                    /* +0 > -0 is preferred by C the standard */
+                    if (x == 0 && y == 0)
+                    {
+                        int xs = signbit(x), ys = signbit(y);
+                        if (xs && !ys)
+                            return x;
+                        if (!xs && ys)
+                            return y;
+                        return y;
+                    }
 
-            if (x > y) {
-                return x;
-            }
-            else {
-                return y;
-            }
-        }
+                    if (x > y) {
+                        return y;
+                    }
+                    else {
+                        return x;
+                    }
+                }
+            #endif
     #endif
 
-    #ifndef fmin
-        static double fmin(double x, double y)
-        {
-            /* z > nan for z != nan is required by C the standard */
-            int xnan = isnan(x), ynan = isnan(y);
+    /* For compilers which aren't MSVC and haven't defined isnan */
+    #ifndef isnan
+        #define isnan(x) ((x) != (x))
+    #endif
 
-            if (xnan || ynan)
-            {
-                if (xnan && !ynan)
-                    return y;
-                if (!xnan && ynan)
-                    return x;
-                return x;
-            }
+    #ifndef isfinite
+        #ifdef _MSC_VER
+            /* MSVC 2015 and newer still don't have everything */
+            #define isfinite(x) _finite((x))
+        #else
+            #define isfinite(x) !isnan((x) + (-x))
+        #endif
+    #endif
 
-            /* +0 > -0 is preferred by C the standard */
-            if (x == 0 && y == 0)
-            {
-                int xs = signbit(x), ys = signbit(y);
-                if (xs && !ys)
-                    return x;
-                if (!xs && ys)
-                    return y;
-                return y;
-            }
+    #ifndef isinf
+        #define isinf(x) (!isfinite(x) && !isnan(x))
+    #endif
 
-            if (x > y) {
-                return y;
-            }
-            else {
-                return x;
-            }
-        }
+    #ifndef fma
+        #define fma(x, y, z) ((x)*(y) + (z))
+    #endif
+
+    /*
+     * portable isnan/isinf; in cmath only available in C++11 and npy_math
+     * versions don't work well (they get undef'd by cmath, see gh-5689)
+     * Implementation based on npy_math.h
+     */
```

Comment:
With respect to this comment, C++11 should be available now on OSX, so this shouldn't be an issue.

@rgommers What do you think?

Reply:
Most recently this was a problem on the Travis CI Ubuntu 12.04 VMs; see https://github.com/scipy/scipy/pull/6653#discussion_r82275237. So maybe not time to mess with things quite yet.

Reply:
Though I'm not thinking through the details carefully right now.

Reply:
According to https://github.com/pypa/manylinux/issues/118#issuecomment-307253138, manylinux has C++ 11, which is the main concern. OSX should definitely have C++11.

Reply:
And also, the main CI is now on trusty.